### DISCUSSION TITLE
Pages not building, environment is missing

### DISCUSSION THREAD

**Discussion by mrhappyma** [community/community#23621] [Other Feature Feedback, Questions, & Ideas]
<p>My site is displaying an error 404. I have it correctly hooked up to github pages. There is an index.html file. I am pulling from the main branch at root. The <code>github-pages</code> environment is missing.</p>
<p>I am having the same problem with another site under ownership of the same organization, except there <em>is</em> a <code>github-pages</code> environment.</p>
<p>For both sites I have disabled and reenabled pages, disabled and reenabled the custom domain, no difference.</p>
<p>Any ideas/ways to fix this?</p>
<p>Thanks,<br>
Dominic</p>

**Comment by jfullstackdev** [comment]
<p>may I ask for the link of the repo so that I can check?</p>

**Comment by mrhappyma** [comment]
<p><a class="mention" href="/xdvrx1">@xdvrx1</a></p>
<p>The first mentioned repo is <a href="https://github.com/vidsites/vidsites" rel="noopener nofollow ugc">vidsites/vidsites</a><br>
The second mentioned repo is <a href="https://github.com/vidsites/toothpaste" rel="noopener nofollow ugc">vidsites/toothpaste</a></p>

**Comment by jfullstackdev** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/b/4bc47539fe71407724bcc214bb91cb2bf937b5db.png" data-download-href="/uploads/short-url/aOgIBtNHJ3yYHvnTeaUe8IflMhR.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078278-e8878e87-1782-4b32-ad42-c84472b5ee84.png" alt="image" data-base62-sha1="aOgIBtNHJ3yYHvnTeaUe8IflMhR" width="690" height="344" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bc47539fe71407724bcc214bb91cb2bf937b5db_2_690x344.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bc47539fe71407724bcc214bb91cb2bf937b5db_2_1035x516.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bc47539fe71407724bcc214bb91cb2bf937b5db_2_1380x688.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bc47539fe71407724bcc214bb91cb2bf937b5db_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1394×696 31.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>you are using a custom domain, that maybe is the problem,</p>
<p>because if your site is already hosting</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/2/1200accd67f8586891efd07435a194b99f7d055e.png" data-download-href="/uploads/short-url/2zg2wkuNIb591o0gLuCeC2Kn47Q.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078279-3c6eefc0-91fb-4caf-8886-55bbab1b9f57.png" alt="image" data-base62-sha1="2zg2wkuNIb591o0gLuCeC2Kn47Q" width="689" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/2/1200accd67f8586891efd07435a194b99f7d055e_2_689x499.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/2/1200accd67f8586891efd07435a194b99f7d055e_2_1033x748.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/2/1200accd67f8586891efd07435a194b99f7d055e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/2/1200accd67f8586891efd07435a194b99f7d055e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1209×876 46.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>the README.md content should appear on your site,</p>
<p>you can give further details so that we can see,</p>
<p>my suggestion is to make sure GitHub Pages is working before you customize the domain, because <a href="https://github.com/vidsites/vidsites" rel="noopener nofollow ugc">https://github.com/vidsites/vidsites</a> is not the root of your GitHub Pages, when it’s deployed it will be <a href="https://vidsites.github.io/vidsites" rel="noopener nofollow ugc">https://vidsites.github.io/vidsites</a></p>
<p>if that is working already then you can proceed to custom domain, that custom domain if DNS is improperly configured can still result to 404,</p>
<p>it will take time to update GitHub Pages before you can see the changes,</p>
<p>so at least we know which one is causing the error,</p>

**Comment by jfullstackdev** [comment]
<p>or try a new repo, put a content on README, enable the GitHub Pages, wait a little bit, and you should be seeing the content on GitHub Pages, even if there is no <code>index.html</code>, because you just want to make sure that GitHub Pages is really working,</p>
<p>if that repo is already being hosted on GitHub Pages, you can put all your website files, and it will surely work</p>

**Comment by mrhappyma** [comment]
<p>I recieve an error 404 even when not using a custom domain. I have waited overnight and still get the same error 404. I have the domain properly configured (I can provide the records if that would be helpful).</p>

**Comment by yoannchaudet** [comment]
<p>Hello <a class="mention" href="/mrhappyma">@mrhappyma</a>!</p>
<p>We are sorry for the inconvenience, this is directly related to a beta feature we shipped yesterday: <a href="https://github.blog/changelog/2021-12-16-github-pages-using-github-actions-for-builds-and-deployments-for-public-repositories/" class="inline-onebox">GitHub Pages: using GitHub Actions for builds and deployments for public repositories | GitHub Changelog</a></p>
<p>Your repository has Actions disabled and this is causing the builds not to happen at all currently.</p>
<p>This is being worked on now and should be resolved within a few hours.</p>
<p>As a mitigation (if you like and temporarily), you can just enable Actions in the repository.</p>

**Comment by mrhappyma** [comment]
<p>Oh, that sorta half makes sense!</p>
<p>tysm for letting me know</p>

**Comment by pacahon** [comment]
<p>Hello <a href="https://github.community/u/yoannchaudet">yoannchaudet</a>!<br>
I see a repo.actions_enabled record  in the org audit log from a person who didn’t do it (she confirmed it). Is it related to this feature?</p>

**Comment by yoannchaudet** [comment]
<p>Hello <a class="mention" href="/pacahon">@pacahon</a>,</p>
<p>That does not sound to me like it can be related. But I suggest reaching out to <a href="https://support.github.com/contact" class="inline-onebox">GitHub Support</a> which will be able to investigate that for you.</p>

**Comment by mrhappyma** [comment]
<p>Hi <a class="mention" href="/yoannchaudet">@yoannchaudet</a>,</p>
<p>My second site is still not loading…</p>

**Comment by yoannchaudet** [comment]
<p>It loads just fine for me here:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/c/ec468a0c1d043f9d07e35fc38cfa820b9473ef72.jpeg" data-download-href="/uploads/short-url/xIbOnT9CG1KIoQlKVejKQzTagTw.jpeg?dl=1" title="image"><img src="https://user-images.githubusercontent.com/14911070/181078284-2a806bb9-1b41-478c-bd93-8fdda3502aa1.jpeg" alt="image" data-base62-sha1="xIbOnT9CG1KIoQlKVejKQzTagTw" width="690" height="479" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec468a0c1d043f9d07e35fc38cfa820b9473ef72_2_690x479.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec468a0c1d043f9d07e35fc38cfa820b9473ef72_2_1035x718.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/c/ec468a0c1d043f9d07e35fc38cfa820b9473ef72.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec468a0c1d043f9d07e35fc38cfa820b9473ef72_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1276×886 49.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mrhappyma** [comment]
<p>Hmmm…<br>
Well now it is working for me.</p>
<p>Thanks?</p>

**Comment by haaami01** [comment]
<p>welcome I also get error code 404, already. I tried everything to make it work. But it’s been almost six months. * :thinking: :flushed:</p>

**Comment by jfullstackdev** [comment]
<p>I am checking at your profile, I can’t see any repo that you are hosting on GitHub Pages, if there is, can you give me the link so that I can check it,</p>

**Comment by haaami01** [comment]
<p><a href="https://github.com/haaami01" rel="noopener nofollow ugc"></a><a href="https://github.com/haaami01" rel="noopener nofollow ugc">https:// github.com/haaami01</a><br>
Hi here’s my profile page so you can see it.</p>
<blockquote>
<p>– Eredeti üzenet –</p>
<p><strong>Feladó:</strong> <a href="mailto:githubcommunity@discoursemail.com">xdvrx1 via GitHub Community</a><a href="mailto:githubcommunity@discoursemail.com">githubcommunity@discoursemail.com</a><br>
<strong>Címzett:</strong> <a href="mailto:szabirepublic01@citromail.hu">szabirepublic01@citromail.hu</a><br>
<strong>Elküldve:</strong> 2021. december 26. 5:0<br>
<strong>Tárgy :</strong> [GitHub Community] [GitHub Help/GitHub Pages] Az oldalak nem épülnek, hiányzik a környezet</p>
<blockquote>
<div class="md-table">
<table>
<thead>
<tr>
<th><img src="https://sea1.discourse-cdn.com/github/user_avatar/github.community/xdvrx1/45/76590_2.png" title="xdvrx1" width="45" height="45"></th>
<th>
<a href="https://github.community/u/xdvrx1">xdvrx1 </a><br>december 26</th>
</tr>
</thead>
<tbody></tbody>
</table>
</div>
</blockquote>
<p>Megnézem a profilodat, nem látok semmilyen repot, amit a GitHub oldalakon tárolsz, ha van, megadnád a linket, hogy ellenőrizhessem?</p>
<hr>
<p><a href="https://github.community/t/pages-not-building-environment-is-missing/218173/15">Látogassa meg a Témát</a> vagy válaszoljon erre az e-mailre.</p>
<p>Ha le szeretne iratkozni ezekről az e-mailekről, <a href="https://github.community/email/unsubscribe/e6627570f482e0333cf2973759c1e72d38155355fe73bc00081ad1ec47b79ecd">kattintson ide</a> .</p>
</blockquote>
<p>Citromail.hu levelezõ rendszerbõl küldve<br>
<a href="https://www.citromail.hu/" rel="noopener nofollow ugc">Lépj be</a> vagy <a href="https://auth.citromail.hu/regisztracio/" rel="noopener nofollow ugc">regisztrálj</a></p>
